projects/rolling-statistics/rolling_statistics.py

```diff
@@ -0,0 +1,699 @@
+        #
+        # rolling_df[rolling_prefixes[i]+"_cgm_25percentile"] = \
+        #    rolling_window.quantile(0.25)
+        # rolling_df[rolling_prefixes[i]+"_cgm_50percentile"] = \
+        #    rolling_window.quantile(0.5)
+        # rolling_df[rolling_prefixes[i]+"_cgm_75percentile"] = \
+        #    rolling_window.quantile(0.75)
+        # rolling_df[rolling_prefixes[i]+"_cgm_90percentile"] = \
+        #    rolling_window.quantile(0.9)
+        # rolling_df[rolling_prefixes[i]+"_cgm_max"] = rolling_window.max()
+        # rolling_df[rolling_prefixes[i]+"_IQR"] = \
+        #    rolling_df[rolling_prefixes[i]+"_cgm_75percentile"]\
+        #    - rolling_df[rolling_prefixes[i]+"_cgm_25percentile"]
+
+        rolling_df[rolling_prefixes[i]+"_events-below54"] = \
+            df["event-below54"].rolling(rolling_points[i],
+                                        min_periods=rolling_min[i]).sum()
+        rolling_df[rolling_prefixes[i]+"_avg-dur-below54"] = df["dur-below54"].rolling(rolling_points[i], min_periods=rolling_min[i]).sum()/rolling_df[rolling_prefixes[i]+"_events-below54"]
+        rolling_df[rolling_prefixes[i]+"_events-below70"] = df["event-below70"].rolling(rolling_points[i] ,min_periods=rolling_min[i]).sum()
+        rolling_df[rolling_prefixes[i]+"_avg-dur-below70"] = df["dur-below70"].rolling(rolling_points[i], min_periods=rolling_min[i]).sum()/rolling_df[rolling_prefixes[i]+"_events-below70"]
+        rolling_df[rolling_prefixes[i]+"_events-above140"] = df["event-above140"].rolling(rolling_points[i], min_periods=rolling_min[i]).sum()
+        rolling_df[rolling_prefixes[i]+"_avg-dur-above140"] = df["dur-above140"].rolling(rolling_points[i], min_periods=rolling_min[i]).sum()/rolling_df[rolling_prefixes[i]+"_events-above140"]
+        rolling_df[rolling_prefixes[i]+"_events-above180"] = df["event-above180"].rolling(rolling_points[i], min_periods=rolling_min[i]).sum()
+        rolling_df[rolling_prefixes[i]+"_avg-dur-above180"] = df["dur-above180"].rolling(rolling_points[i], min_periods=rolling_min[i]).sum()/rolling_df[rolling_prefixes[i]+"_events-above180"]
+        rolling_df[rolling_prefixes[i]+"_events-above250"] = df["event-above250"].rolling(rolling_points[i], min_periods=rolling_min[i]).sum()
+        rolling_df[rolling_prefixes[i]+"_avg-dur-above250"] = df["dur-above250"].rolling(rolling_points[i], min_periods=rolling_min[i]).sum()/rolling_df[rolling_prefixes[i]+"_events-above250"]
+
+        print(rolling_prefixes[i], ' took {0} seconds.'.format(time.time() - start_time))
+
+    return rolling_df
+
+
+def get_daily_stats(df, daytime_start):
+
+    daily_df = df.copy()
+    daily_df.set_index("est.localTime_rounded", inplace=True)
+
+    # Isolate all statistics to just rows at daytime_start (default 5:55)
+    daily_df = daily_df.at_time(daytime_start)
+
+    # Move time back 6 hours so that each row represents the appropriate day
+    daily_df.index = daily_df.index-dt.timedelta(hours=6)
```

Comment:
I would change this so that it is only giving the date, not date time. However, I would also include the start dateTime and the end dateTime to be specific about what time interval you are talking about (i.e., 2017-01-01 06:00 is the start dateTime and 2017-01-02 05:00 is the endTime for the day = 2017-01-01).